src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SymmetricKeyWrap.cs

```diff
@@ -62,31 +73,41 @@ internal static byte[] TripleDESKeyWrapDecrypt(byte[] rgbKey, byte[] rgbEncrypte
+            TripleDES tripleDES = TripleDES.Create();
             // Assume no padding, use CBC mode
             tripleDES.Padding = PaddingMode.None;
             ICryptoTransform dec1 = tripleDES.CreateDecryptor(rgbKey, s_rgbTripleDES_KW_IV);
-            byte[] temp2 = dec1.TransformFinalBlock(rgbEncryptedWrappedKeyData, 0, rgbEncryptedWrappedKeyData.Length);
-            Array.Reverse(temp2);
-            // Get the IV and temp1
-            byte[] rgbIV = new byte[8];
-            Buffer.BlockCopy(temp2, 0, rgbIV, 0, 8);
-            byte[] temp1 = new byte[temp2.Length - rgbIV.Length];
-            Buffer.BlockCopy(temp2, 8, temp1, 0, temp1.Length);
+            ICryptoTransform dec2 = null;
+            try
+            {
+                byte[] temp2 = dec1.TransformFinalBlock(rgbEncryptedWrappedKeyData, 0, rgbEncryptedWrappedKeyData.Length);
+                Array.Reverse(temp2);
+                // Get the IV and temp1
+                byte[] rgbIV = new byte[8];
+                Buffer.BlockCopy(temp2, 0, rgbIV, 0, 8);
+                byte[] temp1 = new byte[temp2.Length - rgbIV.Length];
+                Buffer.BlockCopy(temp2, 8, temp1, 0, temp1.Length);
 
-            ICryptoTransform dec2 = tripleDES.CreateDecryptor(rgbKey, rgbIV);
-            byte[] rgbWKCKS = dec2.TransformFinalBlock(temp1, 0, temp1.Length);
+                dec2 = tripleDES.CreateDecryptor(rgbKey, rgbIV);
+                byte[] rgbWKCKS = dec2.TransformFinalBlock(temp1, 0, temp1.Length);
 
-            // checksum the key
-            byte[] rgbWrappedKeyData = new byte[rgbWKCKS.Length - 8];
-            Buffer.BlockCopy(rgbWKCKS, 0, rgbWrappedKeyData, 0, rgbWrappedKeyData.Length);
-            using (var sha = SHA1.Create())
+                // checksum the key
+                byte[] rgbWrappedKeyData = new byte[rgbWKCKS.Length - 8];
+                Buffer.BlockCopy(rgbWKCKS, 0, rgbWrappedKeyData, 0, rgbWrappedKeyData.Length);
+                using (var sha = SHA1.Create())
+                {
+                    byte[] rgbCKS = sha.ComputeHash(rgbWrappedKeyData);
+                    for (int index = rgbWrappedKeyData.Length, index1 = 0; index < rgbWKCKS.Length; index++, index1++)
+                        if (rgbWKCKS[index] != rgbCKS[index1])
+                            throw new CryptographicException(SR.Cryptography_Xml_BadWrappedKeySize);
+                    return rgbWrappedKeyData;
+                }
+            }
+            finally
             {
-                byte[] rgbCKS = sha.ComputeHash(rgbWrappedKeyData);
-                for (int index = rgbWrappedKeyData.Length, index1 = 0; index < rgbWKCKS.Length; index++, index1++)
-                    if (rgbWKCKS[index] != rgbCKS[index1])
-                        throw new CryptographicException(SR.Cryptography_Xml_BadWrappedKeySize);
-                return rgbWrappedKeyData;
+                tripleDES.Dispose();
+                dec1.Dispose();
+                dec2.Dispose();
```

Comment:
Sure, it'd work.  But I offered the stylistic choice of the try finally to avoid triple nesting (since each object requires work to be done after the prior one's creation, so it can't be a stacked using, like

```C#
using (TripleDES tripleDES = TripleDES.Create())
using (ICryptoTransform dec1 = tripleDES.CreateDecryptor(key, iv))
using (ICryptoTransform dec2 = tripleDES.CreateDecryptor(key2, iv2))
{
    all_the_stuff();
}
```


```C#
using (TripleDES tripleDES = TripleDES.Create())
{
    do_stuff();

    using (ICryptoTransform dec1 = tripleDES.CreateDecryptor(key, iv))
    {
        do_other_stuff();

        using (ICryptoTransform dec2 = tripleDES.CreateDecryptor(key2, iv2))
        {
            // This is getting pretty deep, and probably a bit messy.
            do_yet_other_stuff();
        }
    }
}
```